### ISSUE TITLE
bogus typechecker error only in IntelliJ

### ISSUE THREAD

**Issue by gavinking** [ceylon/ceylon-ide-intellij#618]
I have a class which implements `android.service.media.MediaBrowserService.onLoadChildren()`:

```ceylon
    shared actual void onLoadChildren(String parentMediaId,
            MediaBrowserService.Result<List<MediaBrowser.MediaItem>> result) { ... }
```

This results in the nonsense error:

> type of parameter 'result' of 'onLoadChildren' declared by 'MusicService' is different to type of corresponding parameter 'result' of refined member 'onLoadChildren' of 'MediaBrowserService': 'MediaBrowserService.Result<List<MediaBrowser.MediaItem>>' is not exactly 'Result<List<MediaBrowser.MediaItem>>?'

But the error only occurs inside Android Studio. It doesn't happen from the Gradle build.

You can see the error in this project: <https://github.com/gavinking/UniversalMusicPlayer>

**Comment by bjansen** [comment]
Simpler test case:

```java
public abstract class MediaBrowserService {
    public class Result<T> {}
    abstract void fun(Result<String> list);
}
```
```ceylon
import java.lang { String }

class Cl() extends MediaBrowserService() {
    shared actual void fun(Result<String>? list) {}
}
```

> type of parameter 'list' of 'fun' declared by 'Cl' is different to type of corresponding parameter 'list' of refined member 'fun' of 'MediaBrowserService': 'MediaBrowserService.Result<String>?' is not exactly 'Result<String>?'

**Comment by bjansen** [comment]
```java
    // TODO
    qualifyingType => null; //enclosing;
```

🙄

**Comment by gavinking** [comment]
ops!

**Comment by gavinking** [comment]
@bjansen I was trying to figure this one out and perhaps the answer is to use `psi.accept(PsiTypeVisitor)` to get the qualifying type? WDYT?

**Comment by bjansen** [comment]
I was also looking for a solution the other day, but didn't notice that visitor. Are you talking about [this one](https://github.com/JetBrains/intellij-community/blob/master/java/java-psi-api/src/com/intellij/psi/PsiTypeVisitor.java)? It doesn't look like it does anything useful in our case...

**Comment by gavinking** [comment]
Yeah that one. Oh I thought maybe it visits the constituent types of a type expression.
